File: backend/Python/models/nba/nbateam.py

from models.nba.nbaplayer import NBAPlayer
from dataclasses import dataclass, field
from models.nba.defense_rank import DefensiveRanking
@dataclass
class NBATeam:
    id: str
    full_name: str
    abbreviation: str
    nickname: str
    city: str
    state: str
    defense_rank: list[DefensiveRanking] = field(default_factory=list)
    
    def set_roster(self, roster: list[NBAPlayer]):
        self.roster = roster
        
    def __post_init__(self):
        self.id = str(self.id)
    
    def has_roster(self):
        return hasattr(self, 'roster') and self.roster is not None and len(self.roster) > 0

    def set_defense_rank(self, defense_rank: DefensiveRanking):
       
        if any(dr.playerPosition == defense_rank.playerPosition for dr in self.defense_rank):
            print("already in")
        else:
            self.defense_rank.append(defense_rank)
    
    def has_defense_rank(self):
        return hasattr(self, 'defense_rank') and self.defense_rank is not None and len(self.defense_rank) > 0

File: backend/Go/models/NBATeam.go
package models

type NBATeam struct {
	ID           string      `json:"id"`
	FullName     string      `json:"full_name"`
	Abbreviation string      `json:"abbreviation"`
	Nickname     string      `json:"nickname"`
	City         string      `json:"city"`
	State        string      `json:"state"`
	Roster       []NBAPlayer `json:"roster"`
}

// set roster of team
func (team *NBATeam) SetRoster(roster []NBAPlayer) {
	team.Roster = roster
}

// NewNBATeam is a constructor for NBATeam
func NewNBATeam(id, fullName, abbreviation, nickname, city, state string) *NBATeam {
	return &NBATeam{
		ID:           id,
		FullName:     fullName,
		Abbreviation: abbreviation,
		Nickname:     nickname,
		City:         city,
		State:        state,
	}
}
